zkvm/tests/zkvm.rs

```diff
@@ -0,0 +1,72 @@
+use bulletproofs::r1cs;
+use bulletproofs::{BulletproofGens, PedersenGens};
+use curve25519_dalek::ristretto::CompressedRistretto;
+use curve25519_dalek::scalar::Scalar;
+use merlin::Transcript;
+use std::collections::VecDeque;
+
+use zkvm::*;
+
+fn issue_contract(qty: u64, flv: Scalar, pred: &Predicate) -> Vec<Instruction> {
+    vec![
+        // pushdata qty [qty]
+        Instruction::Push(Commitment::from(CommitmentWitness::from_secret(qty)).into()),
```

Comment:
We can skip `Commitment::from(...).into()` entirely if we change this definition from:

```rust
impl From<Commitment> for DataWitness {
    fn from(x: Commitment) -> Self {
        DataWitness::Commitment(Box::new(x))
    }
}
```

into:

```rust
impl<T> From<T> for DataWitness
where
    T: Into<Commitment>,
{
    fn from(x: T) -> Self {
        DataWitness::Commitment(Box::new(x.into()))
    }
}
```

Then anything convertible into `Commitment` (including itself) will be convertible into DataWitness, and per similar definition for Data, it will be convertible to Data.

Reply:
it seems like we can only have one such conversion, since this I get this error message:
```
conflicting implementations of trait `std::convert::From<_>` for type `types::DataWitness`:
```
where the definition for `Commitment` conflicts with the definition for `ScalarWitness`